radixdlt/src/main/java/com/radixdlt/consensus/liveness/PendingNewViews.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus.liveness;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+import com.radixdlt.SecurityCritical;
+import com.radixdlt.consensus.NewView;
+import com.radixdlt.consensus.View;
+import com.radixdlt.consensus.validators.ValidationState;
+import com.radixdlt.consensus.validators.ValidatorSet;
+import com.radixdlt.crypto.ECDSASignature;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.crypto.Hash;
+import com.radixdlt.utils.Longs;
+
+/**
+ * Manages pending votes for various vertices.
+ * <p>
+ * This class is NOT thread-safe.
+ * <p>
+ * This class is security critical (signature checks, validator set membership checks).
+ */
+@NotThreadSafe
+@SecurityCritical
+public final class PendingNewViews {
+	private static final Logger log = LogManager.getLogger();
+
+	private final Map<View, ValidationState> newViewState = Maps.newHashMap();
+	private final Map<ECPublicKey, View> previousNewView = Maps.newHashMap();
+
+	/**
+	 * Inserts a vote for a given vertex, attempting to form a quorum certificate for that vertex.

```

Comment:
This too would be view, not vertex, right?

Reply:
Fixed.